### PR TITLE
First pass at improving analyzer docs

### PR DESCRIPTION
**Pull request by clintongormley**
I've rewritten the intro to analyzers plus the docs for all analyzers to provide working examples.

I've also removed:
- analyzer aliases (see #18244)
- analyzer versions (see #18267)
- snowball analyzer (see #8690)

Next steps will be tokenizers, token filters, char filters

### CODE REVIEW

File: docs/reference/analysis.asciidoc [hunk 1]
```diff
@@ -3,68 +3,113 @@
...
+up the analyzer to use for each field.
+
+The analyzer to use to search a particular field is determined by
+looking for:
+
+* An `analyzer` specified in the query itself.
+* The <<search-analyzer,`search_analyzer`>> mapping parameter.
+* The <<analyzer,`analyzer`>> mapping parameter.
+* An analyzer in the index settings called `default_search`.
+* An analyzer in the index settings called `search`.
```
Comment (jpountz) [inline docs/reference/analysis.asciidoc]: I am curious why we have both `search` and `default_search`?

File: docs/reference/analysis/analyzers.asciidoc [hunk 2]
```diff
@@ -1,67 +1,60 @@
...
-    analyzer :
-      default :
-        tokenizer : keyword
---------------------------------------------------
+<<analysis-standard-analyzer,Standard Analyzer>>::
 
-[float]
-[[aliasing-analyzers]]
-=== Aliasing Analyzers
+The `standard` analyzer divides text into terms on word bounaries, as defined
```
Comment (jpountz) [inline docs/reference/analysis/analyzers.asciidoc]: s/bounaries/boundaries/

**Comment by jpountz** [comment]
This looks great!

File: docs/reference/analysis.asciidoc [hunk 3]
```diff
@@ -3,68 +3,113 @@
...
+      }
+    }
+  }
+}
+-------------------------
+// CONSOLE
+
+At index time, if no `analyzer` has been specified, it looks for an analyzer
+in the index settings called `default`.  Failing that, it defaults to using
+the <<analysis-standard-analyzer,`standard` analyzer>>.
```
Comment (dadoonet) [inline docs/reference/analysis.asciidoc]: May be we can add an example of what the standard analyzer will do with `"The QUICK brown foxes jumped over the lazy dog!"` as you did for the english analyzer?

Comment (clintongormley) [inline docs/reference/analysis.asciidoc]: I've added examples on each of the analyzer pages.

File: docs/reference/analysis.asciidoc (hunk 1 above)
Comment (clintongormley) [inline docs/reference/analysis.asciidoc]: Because it's incorrect :)  `search` should be `default`

**Comment by clintongormley** [comment]
thanks @jpountz - i've fixed those issues and will merge

**Pull request merged**